File: OsmAnd/src/net/osmand/plus/settings/enums/MetricsConstants.java
package net.osmand.plus.settings.enums;

import android.content.Context;

import androidx.annotation.StringRes;

import net.osmand.plus.R;

public enum MetricsConstants {
	KILOMETERS_AND_METERS(R.string.si_km_m, "km-m"),
	MILES_AND_FEET(R.string.si_mi_feet, "mi-f"),
	MILES_AND_METERS(R.string.si_mi_meters, "mi-m"),
	MILES_AND_YARDS(R.string.si_mi_yard, "mi-y"),
	NAUTICAL_MILES_AND_METERS(R.string.si_nm_mt, "nm-m"),
	NAUTICAL_MILES_AND_FEET(R.string.si_nm_ft, "nm-f");

	private final int key;
	private final String ttsString;

	MetricsConstants(int key, String ttsString) {
		this.key = key;
		this.ttsString = ttsString;
	}

	public String toHumanString(Context ctx) {
		return ctx.getString(key);
	}

	public String toTTSString() {
		return ttsString;
	}

	public boolean shouldUseFeet() {
		return this == MILES_AND_FEET || this == MILES_AND_YARDS || this == NAUTICAL_MILES_AND_FEET;
	}

}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/settings/enums/MetricsConstants.kt
package net.osmand.shared.settings.enums

import net.osmand.shared.units.LengthUnits
import net.osmand.shared.units.SpeedUnits
import net.osmand.shared.util.Localization

enum class MetricsConstants(private val key: String, private val ttsString: String) {
	KILOMETERS_AND_METERS("si_km_m", "km-m"),
	MILES_AND_FEET("si_mi_feet", "mi-f"),
	MILES_AND_METERS("si_mi_meters", "mi-m"),
	MILES_AND_YARDS("si_mi_yard", "mi-y"),
	NAUTICAL_MILES_AND_METERS("si_nm_mt", "nm-m"),
	NAUTICAL_MILES_AND_FEET("si_nm_ft", "nm-f");

	fun toHumanString() = Localization.getString(key)

	fun toTTSString() = ttsString

	fun shouldUseFeet(): Boolean {
		return this == MILES_AND_FEET || (this == MILES_AND_YARDS) || (this == NAUTICAL_MILES_AND_FEET)
	}

	fun getDistanceUnit(): LengthUnits {
		return when(this) {
			MILES_AND_METERS, MILES_AND_FEET, MILES_AND_YARDS -> LengthUnits.MILES
			NAUTICAL_MILES_AND_FEET, NAUTICAL_MILES_AND_METERS -> LengthUnits.NAUTICAL_MILES
			KILOMETERS_AND_METERS -> LengthUnits.KILOMETERS
		}
	}

	fun getSpeedUnit(): SpeedUnits {
		return when(this) {
			MILES_AND_METERS, MILES_AND_FEET, MILES_AND_YARDS -> SpeedUnits.MILES_PER_HOUR
			NAUTICAL_MILES_AND_FEET, NAUTICAL_MILES_AND_METERS -> SpeedUnits.KNOTS
			KILOMETERS_AND_METERS -> SpeedUnits.KILOMETERS_PER_HOUR
		}
	}

	fun getAltitudeMetrics(): AltitudeMetrics {
		return when(this) {
			KILOMETERS_AND_METERS, NAUTICAL_MILES_AND_METERS, MILES_AND_METERS -> AltitudeMetrics.METERS
			else -> AltitudeMetrics.FEET
		}
	}
}